extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/StockMethodsAdder.java

```diff
@@ -134,6 +134,13 @@ private void generateSave(ClassCreator classCreator, FieldDescriptor entityClass
                     BytecodeCreator idValueUnset;
                     BytecodeCreator idValueSet;
                     if (idType instanceof PrimitiveType) {
+                        if (!idType.name().equals(DotNames.PRIMITIVE_LONG)
+                                && !idType.name().equals(DotNames.PRIMITIVE_INTEGER)) {
+                            throw new IllegalArgumentException("Id type of '" + entityDotName + "' is invalid.");
+                        }
+                        if (idType.name().equals(DotNames.PRIMITIVE_LONG)) {
+                            idValue = save.checkCast(idValue, int.class);
```

Comment:
Why this cast?

Reply:
We need to cast into `int` otherwise the bytecode verifier complains about bad operands on the stack

Reply:
I suppose if the long is not a valid int, it won't be zero anyway. That's your rationale?

Reply:
Exactly